File: app/src/main/java/com/suda/datetimewallpaper/service/DateTimeDreamService.java
package com.suda.datetimewallpaper.service;

import android.service.dreams.DreamService;

import android.view.View;
import com.suda.datetimewallpaper.R;

/**
 * @author guhaibo
 * @date 2019/4/9
 */
public class DateTimeDreamService extends DreamService {
    @Override
    public void onAttachedToWindow() {
        super.onAttachedToWindow();
        setFullscreen(true);
        setContentView(R.layout.dream_service_layout);
        View decorView = getWindow().getDecorView();
        int uiOptions = View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
                | View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY | View.SYSTEM_UI_FLAG_FULLSCREEN;
        decorView.setSystemUiVisibility(uiOptions);
    }
}

File: app/src/main/java/com/suda/datetimewallpaper/service/DateTimeDreamService.kt
package com.suda.datetimewallpaper.service

import android.service.dreams.DreamService
import android.view.View
import com.suda.datetimewallpaper.R
import com.suda.datetimewallpaper.util.SharedPreferencesUtil
import com.suda.datetimewallpaper.view.DateTimeView

/**
 * @author guhaibo
 * @date 2019/4/9
 */
class DateTimeDreamService : DreamService() {
    override fun onAttachedToWindow() {
        super.onAttachedToWindow()
        isFullscreen = true
        setContentView(R.layout.layout_dream_service)
        val dateTimeView = findViewById<View>(R.id.dtv) as DateTimeView
        val sharedPreferencesUtil = SharedPreferencesUtil(this)
        dateTimeView.resetPaperId(sharedPreferencesUtil.nextWallPaperDream)
        val decorView = window.decorView
        val uiOptions = (View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
                or View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY or View.SYSTEM_UI_FLAG_FULLSCREEN)
        decorView.systemUiVisibility = uiOptions
    }
}
